rskj-core/src/main/java/org/ethereum/rpc/LogFilter.java

```diff
@@ -78,19 +73,48 @@ void onTransaction(Transaction tx, Block b, int txIndex) {
             LogInfo logInfo = receipt.getLogInfoList().get(i);
 
             if (addressesTopicsFilter.matchesExactly(logInfo)) {
-                onLogMatch(logInfo, b, txIndex, receipt.getTransaction(), i);
+                onLogMatch(logInfo, block, txIndex, receipt.getTransaction(), i);
             }
         }
     }
 
-    void onBlock(Block b) {
-        if (addressesTopicsFilter.matchBloom(new Bloom(b.getLogBloom()))) {
-            int txIdx = 0;
+    void onTransaction(Transaction tx, Block block, int txIndex) {
+        TransactionInfo txInfo = blockchain.getTransactionInfo(tx.getHash().getBytes());

```

Comment:
```suggestion
        TransactionInfo txInfo = blockchain.getTransactionInfo(tx.getHash().getBytes(), block.getHash().getBytes());

```

seems we don't need extra methods here - `onBlockFromMainChain()`  and `onTransactionFromMainChain()`? Incorporating that new method `BlockChainImpl::getTransactionInfo(Transaction tx, byte[] blockHash)` you added should be sufficient?

Reply:
Good point. Addressed.